src/main/java/seedu/duke/command/SeeAllIncomesCommand.java

```diff
@@ -0,0 +1,62 @@
+package seedu.duke.command;
+
+import seedu.duke.financial.FinancialEntry;
+import seedu.duke.financial.FinancialList;
+import seedu.duke.financial.Income;
+
+import java.time.LocalDate;
+
+/**
+ * Command to print all incomes recorded in the financial list.
+ */
+public class SeeAllIncomesCommand extends Command {
+    private LocalDate start;
+    private LocalDate end;
+
+    public SeeAllIncomesCommand(LocalDate start, LocalDate end) {
+        this.start = start;
+        this.end = end;
+    }
+
+    /**
+     * Method to determine if an entry should be listed out based on its date and if it is an income.
+     *
+     * @param entry Financial Entry to analyze.
+     * @return true if entry should be listed out, false otherwise.
+     */
+    private boolean shouldBeIncluded(FinancialEntry entry) {
+        return entry instanceof Income && (end == null || entry.getDate().isBefore(end))
+                && (start == null || entry.getDate().isAfter(start));
```

Comment:
Maybe you could avoid such [complicated expressions](https://nus-cs2113-ay2425s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-complicated-expressions) and look to refactor it to make it simpler to read